test/FunctionalTests/Microsoft.OData.Core.Tests/Json/MockAsyncOnlyJsonWriter.cs

```diff
@@ -0,0 +1,151 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="MockAsyncOnlyJsonWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.OData.Edm;
+using Microsoft.OData.Json;
+
+namespace Microsoft.OData.Tests.Json
+{
+    /// <summary>
+    /// A mock implementation of <see cref="IJsonWriterAsync"/> that
+    /// does not implement the synchronous interface <see cref="IJsonWriter"/>.
+    /// This is used for testing purposes. Most of the methods have not been implemented.
+    /// </summary>
+    internal class MockAsyncOnlyJsonWriter : IJsonWriterAsync
+    {
+        public Task EndArrayScopeAsync()

```

Comment:
nit: You could keep the class shorter by using the shorter form, e.g., public void EndArrayScopeAsync() => new NotImplementedException();

Reply:
Thanks for the tip. I'm looking at all the methods I'd have to do this for and wondering whether I should 🥲

Reply:
Done!